api/utils/ec2.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"regexp"
+)
+
+// EC2 Node IDs are {AWS account ID}-{EC2 resource ID} eg:
+//
+//	123456789012-i-1234567890abcdef0
+//
+// AWS account ID is always a 12 digit number, see
+//
+//	https://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html
+//
+// EC2 resource ID is i-{8 or 17 hex digits}, see
+//
+//	https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/resource-ids.html
+var ec2NodeIDRE = regexp.MustCompile("^[0-9]{12}-i-[0-9a-f]{8,}$")
```

Comment:
Sounds right.  This isn't actually new though, I just messed up the diff (it's actually just a moved/renamed file).  Will probably leave as-is for now, just because I'm reticent to make too many separate changes to how routing works at the same time, and I don't actually know the full context for this regex.